src/hal-backend-ml-vivante.cc

```diff
@@ -216,45 +216,50 @@ _helper_parse_tensor_attributes (JsonObject *tensor_obj, vsi_nn_tensor_attr_t *v
 static int
 _json_create_neural_network (vivante_handle_s *self)
 {
-  g_autofree gchar *json_string = NULL;
-  g_autoptr (GError) err = NULL;
-  g_autoptr (JsonParser) parser = NULL;
+  gchar *json_string = NULL;
+  GError *err = NULL;
+  JsonParser *parser = NULL;
   JsonNode *root_node = NULL;
   JsonObject *root_obj = NULL;
   JsonArray *input_array = NULL, *output_array = NULL;
   guint input_tensors_num = 0, output_tensors_num = 0;
   vsi_nn_node_t *node = NULL;
+  int ret = HAL_ML_ERROR_RUNTIME_ERROR;
```

Comment:
May move const guints here to prevent wrong goto
```suggestion
  const guint node_num = 1U; /* single NBG node */
  const guint const_tensors_num = 0U; /** @todo support this */
  guint normal_tensors_num = 0U;
  guint virtual_tensors_num = 0U;
  int ret = HAL_ML_ERROR_RUNTIME_ERROR;
```